pkg/rehearse/jobs.go

```diff
@@ -289,9 +293,11 @@ func (e *Executor) waitForJobs(jobs sets.String, selector string) (bool, error)
 			switch pj.Status.State {
 			case pjapi.FailureState, pjapi.AbortedState, pjapi.ErrorState:
 				e.loggers.Job.WithFields(fields).Error("Job failed")
+				e.Metrics.RecordFailedRehearsal(pj)
```

Comment:
We have to change the logic here and avoid passing an `ExecutionMetrics`.  What are you doing here is logical and correct but the whole `waitForJobs` logic is not so good. (that's why I tried to do that refactor and pass the result in the channels here). Of course, we have to discuss this in one of our meetings.